test/sql-luatest/show_create_table_test.lua

```diff
@@ -0,0 +1,456 @@
+local server = require('luatest.server')
+local t = require('luatest')
+
+local g = t.group()
+
+g.before_all(function()
+    g.server = server:new({alias = 'master'})
+    g.server:start()
+end)
+
+g.after_all(function()
+    g.server:stop()
+end)
+
+g.test_show_create_table_one = function()
+    g.server:exec(function()
+        local _, err = box.execute([[SHOW CREATE TABLE t;]])
+        t.assert_equals(err.message, [[Space 'T' does not exist]])
+
+        box.execute([[CREATE TABLE t(i INT PRIMARY KEY, a INT);]])
+        local ret = box.execute([[SHOW CREATE TABLE t;]])
+        local res = [[CREATE TABLE "T"("I" INTEGER NOT NULL,"A" INTEGER,]]..
+                    [[CONSTRAINT "pk_unnamed_T_1" PRIMARY KEY("I"));]]
+        t.assert_equals(ret.rows[1][1]:gsub('\n', ''), res)
+        box.execute([[DROP TABLE t;]])
+
+        local sql = [[CREATE TABLE "T"("I" INTEGER NOT NULL,]]..
+                    [["A" STRING CONSTRAINT "C2" CHECK(a != 'asd') ]]..
+                    [[CONSTRAINT "C3" REFERENCES "T"("I"),]]..
+                    [["C" UUID NOT NULL DEFAULT(uuid()),]]..
+                    [[CONSTRAINT "C0" PRIMARY KEY("I", "C"),]]..
+                    [[CONSTRAINT "C1" UNIQUE("A"),]]..
+                    [[CONSTRAINT "C4" UNIQUE("A", "C"),]]..
+                    [[CONSTRAINT "C6" CHECK(i * a < 100),]]..
+                    [[CONSTRAINT "C5" FOREIGN KEY("I", "A") ]]..
+                    [[REFERENCES "T"("A", "C")) WITH ENGINE = 'vinyl';]]
+        box.execute(sql)
+        ret = box.execute([[SHOW CREATE TABLE t;]])
+        t.assert_equals(ret.rows[1][1]:gsub('\n', ''), sql)
+        box.execute([[DROP TABLE t;]])
+    end)
+end
+
+g.test_show_create_table_all = function()
+    g.server:exec(function()
+        local res = box.execute([[SHOW CREATE TABLE;]])
+        t.assert_equals(res.rows, {})
+
+        box.execute([[CREATE TABLE t1(i INT PRIMARY KEY, a INT);]])
+        box.execute([[CREATE TABLE t2(i INT PRIMARY KEY, a INT);]])
+        box.execute([[CREATE TABLE t3(i INT PRIMARY KEY, a INT);]])
+        local ret = box.execute([[SHOW CREATE TABLE;]])
+        local res1 = [[CREATE TABLE "T1"("I" INTEGER NOT NULL,"A" INTEGER,]]..
+                     [[CONSTRAINT "pk_unnamed_T1_1" PRIMARY KEY("I"));]]
+        local res2 = [[CREATE TABLE "T2"("I" INTEGER NOT NULL,"A" INTEGER,]]..
+                     [[CONSTRAINT "pk_unnamed_T2_1" PRIMARY KEY("I"));]]
+        local res3 = [[CREATE TABLE "T3"("I" INTEGER NOT NULL,"A" INTEGER,]]..
+                     [[CONSTRAINT "pk_unnamed_T3_1" PRIMARY KEY("I"));]]
+        t.assert_equals(#ret.rows, 3)
+        t.assert_equals(ret.rows[1][1]:gsub('\n', ''), res1)
+        t.assert_equals(ret.rows[2][1]:gsub('\n', ''), res2)
+        t.assert_equals(ret.rows[3][1]:gsub('\n', ''), res3)
```

Comment:
You might want to consider this simplification (you do gsub ~35 times in this file right now):

```Diff
@@ -14,14 +14,26 @@ end)
 
 g.test_show_create_table_one = function()
     g.server:exec(function()
-        local _, err = box.execute([[SHOW CREATE TABLE t;]])
+        rawset(_G, 'test_exec', function(...)
+            local res = box.execute()
+            t.assert(res ~= nil)
+            for i = 1, #res.rows do
+                local field = res.rows[i][1]
+                if field ~= nil then
+                    res.rows[i]:update({{'=', 1, field:gsub('\n', '')}})
+                end
+            end
+            return res
+        end)
+
+        local _, err =  box.execute([[SHOW CREATE TABLE t;]])
         t.assert_equals(err.message, [[Space 'T' does not exist]])
 
         box.execute([[CREATE TABLE t(i INT PRIMARY KEY, a INT);]])
-        local ret = box.execute([[SHOW CREATE TABLE t;]])
+        local ret = _G.test_exec([[SHOW CREATE TABLE t;]])
         local res = [[CREATE TABLE "T"("I" INTEGER NOT NULL,"A" INTEGER,]]..
                     [[CONSTRAINT "pk_unnamed_T_1" PRIMARY KEY("I"));]]
-        t.assert_equals(ret.rows[1][1]:gsub('\n', ''), res)
+        t.assert_equals(ret.rows[1][1], res)
         box.execute([[DROP TABLE t;]])
 
         local sql = [[CREATE TABLE "T"("I" INTEGER NOT NULL,]]..
@@ -35,8 +47,8 @@ g.test_show_create_table_one = function()
                     [[CONSTRAINT "C5" FOREIGN KEY("I", "A") ]]..
                     [[REFERENCES "T"("A", "C")) WITH ENGINE = 'vinyl';]]
         box.execute(sql)
-        ret = box.execute([[SHOW CREATE TABLE t;]])
-        t.assert_equals(ret.rows[1][1]:gsub('\n', ''), sql)
+        ret = _G.test_exec([[SHOW CREATE TABLE t;]])
+        t.assert_equals(ret.rows[1][1], sql)
         box.execute([[DROP TABLE t;]])
     end)
 end
@@ -49,7 +61,7 @@ g.test_show_create_table_all = function()
         box.execute([[CREATE TABLE t1(i INT PRIMARY KEY, a INT);]])
         box.execute([[CREATE TABLE t2(i INT PRIMARY KEY, a INT);]])
         box.execute([[CREATE TABLE t3(i INT PRIMARY KEY, a INT);]])
-        local ret = box.execute([[SHOW CREATE TABLE;]])
+        local ret = _G.test_exec([[SHOW CREATE TABLE;]])
         local res1 = [[CREATE TABLE "T1"("I" INTEGER NOT NULL,"A" INTEGER,]]..
                      [[CONSTRAINT "pk_unnamed_T1_1" PRIMARY KEY("I"));]]
         local res2 = [[CREATE TABLE "T2"("I" INTEGER NOT NULL,"A" INTEGER,]]..
@@ -57,9 +69,9 @@ g.test_show_create_table_all = function()
         local res3 = [[CREATE TABLE "T3"("I" INTEGER NOT NULL,"A" INTEGER,]]..
                      [[CONSTRAINT "pk_unnamed_T3_1" PRIMARY KEY("I"));]]
         t.assert_equals(#ret.rows, 3)
-        t.assert_equals(ret.rows[1][1]:gsub('\n', ''), res1)
-        t.assert_equals(ret.rows[2][1]:gsub('\n', ''), res2)
-        t.assert_equals(ret.rows[3][1]:gsub('\n', ''), res3)
+        t.assert_equals(ret.rows[1][1], res1)
+        t.assert_equals(ret.rows[2][1], res2)
+        t.assert_equals(ret.rows[3][1], res3)
```
and etc similarly.

Reply:
Instead, I decided to add '\n' in expected result. However, I still added two helper functions.